### PR TITLE
urllinter: moving urllinter in TCE under hack dir

### LINKED ISSUE #2948: Url linter should not fail on certificate error
## Bug Report

A recent patch adding an adopter is failing on the link to their web site with the error:

```
Error: Get "https://terasky.com/": x509: certificate signed by unknown authority
```

Going to the site directly, it looks like their certificate is fine. So this maybe isn't picking up the host's configured certificate authorities.

But after talking this through, in general we should not care about invalid certs, only that the link actually goes somewhere. So the check should set `InsecureSkipVerify` to `true` so these types of errors are ignored.

The source code for this tool should also be moved in to the `community-edition` repo so we can make updates there.

## Expected Behavior

Test should pass if the URL actually goes to somewhere legitimate.

## Steps to Reproduce the Bug

Run local tests against #2944

### PR DESCRIPTION
**Pull request by aman556**
## What this PR does / why we need it

This PR is moving urllinter from [here](https://github.com/JitenPalaparthi/urllinter) to `hack` dir.

Files which are changed in this PR
- `Makefile`: Previously make target for urllinter was installing the urllinter from other repo now we are running it from `hack` dir only.
- `hack/urllinter/main.go`:  The import location for `linturl` is changed in this file.
- `hack/urllinter/go.mod`: The name of module is changed.
- To resolve all the linters errors like unused variables, duplication of code, extra spaces and other errors showed by `golangci-lint` the code is changed a bit.
## Details for the Release Notes (PLEASE PROVIDE)

```release-note
moving urllinter in TCE under hack dir
```

## Which issue(s) this PR fixes

Fixes: #2948 

## Describe testing done for PR

Run `make urllint` and it is running fine.

### CODE REVIEW

**Comment by dvonthenen** [comment]
It looks like there are a number of errors in markdown and etc that need to be addressed

**Comment by aman556** [comment]
Hello @dvonthenen thanks for referring I am working on them will update the PR soon.

**Comment by aman556** [comment]
As Url linter is currently in different repo due to which all the files are new here and reviewing them is a bit difficult so i have forked current repo in my account and pushed these changes there so there all the changes in this PR are visible.
This is the link > https://github.com/JitenPalaparthi/urllinter/compare/main...aman556:review-branch-urllinter

**Comment by karuppiah7890** [comment]
We could take care of #2948 too in this PR 😄 Or it could be a separate PR

**Comment by aman556** [comment]
Thanks @karuppiah7890 for mentioning this issue. I will prefer to do in separate PR.

File: hack/urllinter/pkg/lint/lint.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+	count := 0
+	for key := range llc.LinkMap {
+		count++
+		fmt.Println("Currently checking ", count, " url(s) out of ", len(llc.LinkMap))
+		if !IsURL(key) {
+			isFatal = true
+			llc.OnFail("Invalid URL", key)
+			continue
+		}
+		cli := &http.Client{}
```
Comment (karuppiah7890) [inline hack/urllinter/pkg/lint/lint.go:182]: For #2948 

We need to introduce a configuration in `http.Client{}` like this -

```golang
tr := &http.Transport{
    TLSClientConfig: {
      InsecureSkipVerify: true
    }
}
cli := &http.Client{Transport: tr}
```

References - 
- https://pkg.go.dev/net/http#pkg-overview
- https://pkg.go.dev/net/http#Transport
- https://pkg.go.dev/crypto/tls#Config

Comment (karuppiah7890) [reply hack/urllinter/pkg/lint/lint.go:182]: cc @aman556

Comment (aman556) [reply hack/urllinter/pkg/lint/lint.go:182]: I have triggered the URL linter after adding the failing URL once it complete i will try the above suggestion and check that it is working or not.

Comment (aman556) [reply hack/urllinter/pkg/lint/lint.go:182]: after adding '"crypto/tls"` library and adding this
```tr := &http.Transport{
        TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
    	}

		cli := &http.Client{Transport: tr}```
issue got resolved. I am updating this PR.

File: hack/urllinter/go.mod [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module github.com/vmware-tanzu/community-edition/hack/urllinter
+
+go 1.16
```
Comment (stmcginnis) [inline hack/urllinter/go.mod:3]: This should be updated to go 1.17 now.

**Review by stmcginnis — COMMENTED** [review]
One minor thing pointed out. Could you squash your commits?

Thanks!

**Review by stmcginnis — APPROVED** [review]
Looks good, thanks!

**Review by stmcginnis — COMMENTED** [review]
Hmm, looks good, but I did a quick test of the URL that was hitting an issue earlier. It looks like it still fails:

```
URL: https://terasky.com
Error: Get "https://terasky.com": x509: certificate signed by unknown authority
```

**Comment by aman556** [comment]
strange I added the same link in one file and it was failing before doing the changes later it passed after making insecureSkipVerify: true. @stmcginnis can you please share the file so that I can try and test at my end.

**Comment by stmcginnis** [comment]
I randomly picked the file `hack/addon-dev/dev-deploy-addon.sh` and added a line of `wget https://terasky.com`.

**Comment by stmcginnis** [comment]
Actually, hold that. Looks like I had an issue with how the code was checked out locally. I cleaned things up and ran again, and it looks like things are working as expected. Sorry about the confusion!

**Review by stmcginnis — APPROVED** [review]

**Comment by aman556** [comment]
NP @stmcginnis Thanks for merging this.

**Pull request merged**